### PR TITLE
SUM-33 SUM-39 | @rebeccahongsf | theme & style basic and pill card

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- Theme & style basic and pill card
- Pill card dependent on https://github.com/SU-SWS/ace-stanfordlagunita/pull/100

# Review By (Date)
- When possible

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Create a basic page with the default and pill cards
3. Spin up your local and review the page
4. Verify that it appears the same as the figma mocks
5. Review code 👾 

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory? **Nope**

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

# Associated Issues and/or People
- SUM-33
- SUM-39

### CODE REVIEW

File: src/lib/drupal/drupal-jsonapi.d.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ export type CardParagraphBehaviors = {
   heading?: "h2" | "h3" | "h4" | "div.su-splash-font"
   hide_heading?: boolean
   link_style?: "action" | "button"
+  card_variant?: "default" | "pill"
```
Comment (pookmish) [inline src/lib/drupal/drupal-jsonapi.d.tsx:22]: I believe it's this 
```suggestion
  card_variant?: "pill"
```

**Review by pookmish — APPROVED** [review]
Just one question/thought. Just confirm or reject and g2g

**Comment by rebeccahongsf** [comment]
Re-requesting review since I rolled up the pill card styles into this PR as well 👀 Adding the the pill card styles is dependent on Marc's work in https://github.com/SU-SWS/ace-stanfordlagunita/pull/100

**Comment by mdyoung3** [comment]
@rebeccahongsf 

I noticed pills without images have text that bleeds out of the border:

Screenshot 2024-05-17 at 6 40 34 PM

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 2]
```diff
@@ -20,18 +22,22 @@ const CardParagraph = ({paragraph, ...props}: Props) => {
...
   const headerTagChoice = (behaviors.su_card_styles?.heading || "h2").split(".", 2);
   const headerTag = headerTagChoice[0]
-  const headerClasses = headerTagChoice[1]?.replace(".", " ").replace("su-font-splash", "text-m2 font-bold")
-
+  const headerClasses = headerTagChoice[1]?.replace(".", " ").replace("su-font-splash", "text-m2 font-normal mb-12") || "font-normal mb-12"
+  const cardVariant = behaviors.su_card_styles?.sum_card_variant;
+  const pillCardBgColor = behaviors.su_card_styles?.sum_card_bg_color_variant;
   const hideHeader = behaviors.su_card_styles?.hide_heading;
 
+  const Element = cardVariant ? PillCard : ImageCard
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:30]: Let's check for the value of the variant. We might have more variants in the future.

File: src/components/paragraphs/stanford-card/card-paragraph.tsx [hunk 3]
```diff
@@ -20,18 +22,22 @@ const CardParagraph = ({paragraph, ...props}: Props) => {
...
   return (
-    <ImageCard
+    <Element
       {...props}
       aria-labelledby={paragraph.suCardHeader ? paragraph.id : undefined}
       imageUrl={image?.url}
       imageAlt={image?.alt}
       videoUrl={videoUrl}
       isArticle={!!paragraph.suCardHeader}
+      bgColor={pillCardBgColor}
```
Comment (pookmish) [inline src/components/paragraphs/stanford-card/card-paragraph.tsx:40]: Does this throw any logs since the image card doesn't take in this prop?

**Review by pookmish — CHANGES_REQUESTED** [review]
Small change requested. Note, I didn't actually view it.

File: src/components/paragraphs/stanford-card/card-paragraph.tsx (hunk 3 above)
Comment (rebeccahongsf) [reply src/components/paragraphs/stanford-card/card-paragraph.tsx:40]: Strangely there are no error throw for the image card 🤔

Comment (pookmish) [reply src/components/paragraphs/stanford-card/card-paragraph.tsx:40]: > Warning: React does not recognize the `bgColor` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `bgcolor` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

**Review by pookmish — CHANGES_REQUESTED** [review]
Visually, this looks great. But we should address the `bgColor` prop. 

I believe changing the variable to `undefined` will solve the issue: `const pillCardBgColor = cardVariant === "pill" ? behaviors.su_card_styles?.sum_card_bg_color_variant : undefined`

File: src/components/paragraphs/stanford-card/card-paragraph.tsx (hunk 3 above)
Comment (pookmish) [reply src/components/paragraphs/stanford-card/card-paragraph.tsx:40]: I was able to see this in the console when I create a simple card, just add the headline field and then leave all other fields alone except choose a bg color.

**Review by pookmish — APPROVED** [review]

**Pull request merged**